components/agent/my-plugins.tsx

```diff
@@ -0,0 +1,128 @@
+import { apiInterceptors, postAgentMy, postAgentUninstall, postAgentUpload } from '@/client/api';
+import { IMyPlugin } from '@/types/agent';
+import { useRequest } from 'ahooks';
+import { Button, Card, Spin, Tag, Tooltip, Upload, UploadProps, message } from 'antd';
+import { useCallback, useState } from 'react';
+import MyEmpty from '../common/MyEmpty';
+import { ClearOutlined, LoadingOutlined, UploadOutlined } from '@ant-design/icons';
+import { useTranslation } from 'react-i18next';
+
+function MyPlugins() {
+  const { t } = useTranslation();
+  const [messageApi, contextHolder] = message.useMessage();
+
+  const [uploading, setUploading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [actionIndex, setActionIndex] = useState<number | undefined>();
+
+  const {
+    data = [],
+    loading,
+    refresh,
+  } = useRequest(async () => {
+    const [err, res] = await apiInterceptors(postAgentMy());
+    setIsError(!!err);
+    return res ?? [];
+  });
+
+  const uninstall = async (name: string, index: number) => {
+    if (actionIndex) return;
+    setActionIndex(index);
+    const [err] = await apiInterceptors(postAgentUninstall(name));
+    message[err ? 'error' : 'success'](err ? 'failed' : 'success');
+    !err && refresh();
+    setActionIndex(undefined);
+  };
+
+  const renderAction = useCallback(
+    (item: IMyPlugin, index: number) => {
+      if (index === actionIndex) {
+        return <LoadingOutlined />;
+      }
+      return (
+        <Tooltip title="Uninstall">
+          <div
+            className="w-full h-full"
+            onClick={() => {
+              uninstall(item.name, index);
+            }}
+          >
+            <ClearOutlined />
+          </div>
+        </Tooltip>
+      );
+    },
+    [actionIndex],
+  );
+
+  const onChange: UploadProps['onChange'] = async (info) => {
+    if (!info) {
+      message.error('Please select the *.zip,*.rar file');
+      return;
+    }
+    if (!/\.(zip|rar)$/.test(info.file.name ?? '')) {
```

Comment:
you have set `accept` props, other types shouldn‘t be selected, does the judge necessary?